src/executor.ts

```diff
@@ -54,19 +54,52 @@ export class Executor {
   readonly #logger: Logger;
   readonly #workerPool: WorkerPool;
   readonly #cache?: Cache;
-  readonly #failureMode: FailureMode;
-  readonly #failureDeferred = new Deferred<void>();
+
+  /** Resolves when the first failure occurs in any script. */
+  readonly #failureOccured = new Deferred<void>();
+  /** Resolves when we decide that new scripts should be cancelled. */
```

Comment:
Here and elsewhere, seems like "should not start new scripts" is clearer than "should cancel new scripts"

Reply:
Done. New terms: `#stopStartingNewScripts`, `StartCancelled` event.